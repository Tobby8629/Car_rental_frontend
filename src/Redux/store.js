import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import CarSlice from './CarSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Cars: CarSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
