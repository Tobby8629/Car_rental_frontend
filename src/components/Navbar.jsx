import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './componentsCss/navbar.css';

function Navbar() {
  const [tray, settray] = useState(true);

  return (
    <>
      <div className="mobile bars">
        <span role="button" onClick={() => settray(false)} onKeyDown={() => settray(false)} tabIndex={0}>
          <i className="fa-solid fa-bars" />
        </span>
      </div>
      <div id="navbar" className={tray ? 'block' : 'show'} role="button" onClick={() => settray(true)} onKeyDown={() => settray(false)} tabIndex={0}>
        <div className="top">
          <h2>CAR RENT</h2>
          <ul>
            <li>
              <Link to="/">Car</Link>
            </li>
            <li>
              <Link to="reserve">Reserve</Link>
            </li>
            <li>
              <Link to="myreserve">My Reservations</Link>
            </li>
            <li>
              <Link to="add">Add Car</Link>
            </li>
            <li>
              <Link to="delete">Delete Car</Link>
            </li>
            <li>
              <Link to="login">sign in</Link>
            </li>
          </ul>
        </div>
        <div className="socials">
          <ul>
            <li><i className="fa-brands fa-twitter" /></li>
            <li><i className="fa-brands fa-facebook-f" /></li>
            <li><i className="fa-brands fa-pinterest-p" /></li>
            <li><i className="fa-brands fa-google-plus-g" /></li>
          </ul>
        </div>
        <div className="mobile cancel" role="button" onClick={() => settray(true)} onKeyDown={() => settray(true)} tabIndex={0}>
          <i className="fa-solid fa-xmark" />
        </div>
      </div>
    </>
  );
}

export default Navbar;
