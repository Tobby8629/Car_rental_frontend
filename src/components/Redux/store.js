import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
