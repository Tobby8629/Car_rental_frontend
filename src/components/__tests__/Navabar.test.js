import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../Redux/store';
import Navbar from '../Navbar';

describe('Navbar Component Testing', () => {
  test('Navebar Screenshot', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Navbar />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders Navbar component text', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    const linkElement = tree.getByText('CAR RENT');
    expect(linkElement).toBeInTheDocument();
  });
});
