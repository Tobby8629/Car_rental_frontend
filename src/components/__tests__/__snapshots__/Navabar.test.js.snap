// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Testing Navebar Screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mobile bars"
      >
        <span
          role="button"
          tabindex="0"
        >
          <i
            class="fa-solid fa-bars"
          />
        </span>
      </div>
      <div
        class="block"
        id="navbar"
        role="button"
        tabindex="0"
      >
        <div
          class="top"
        >
          <h2>
            CAR RENT
          </h2>
          <ul>
            <li>
              <a
                href="/"
              >
                Car
              </a>
            </li>
            <li>
              <a
                href="/reserve"
              >
                Reserve
              </a>
            </li>
            <li>
              <a
                href="/myreserve"
              >
                My Reservations
              </a>
            </li>
            <li>
              <a
                href="/add"
              >
                Add Car
              </a>
            </li>
            <li>
              <a
                href="/delete"
              >
                Delete Car
              </a>
            </li>
            <li>
              <a
                href="/login"
              >
                sign out
              </a>
            </li>
          </ul>
        </div>
        <div
          class="socials"
        >
          <ul>
            <li>
              <i
                class="fa-brands fa-twitter"
              />
            </li>
            <li>
              <i
                class="fa-brands fa-facebook-f"
              />
            </li>
            <li>
              <i
                class="fa-brands fa-pinterest-p"
              />
            </li>
            <li>
              <i
                class="fa-brands fa-google-plus-g"
              />
            </li>
          </ul>
        </div>
        <div
          class="mobile cancel"
          role="button"
          tabindex="0"
        >
          <i
            class="fa-solid fa-xmark"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mobile bars"
    >
      <span
        role="button"
        tabindex="0"
      >
        <i
          class="fa-solid fa-bars"
        />
      </span>
    </div>
    <div
      class="block"
      id="navbar"
      role="button"
      tabindex="0"
    >
      <div
        class="top"
      >
        <h2>
          CAR RENT
        </h2>
        <ul>
          <li>
            <a
              href="/"
            >
              Car
            </a>
          </li>
          <li>
            <a
              href="/reserve"
            >
              Reserve
            </a>
          </li>
          <li>
            <a
              href="/myreserve"
            >
              My Reservations
            </a>
          </li>
          <li>
            <a
              href="/add"
            >
              Add Car
            </a>
          </li>
          <li>
            <a
              href="/delete"
            >
              Delete Car
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              sign out
            </a>
          </li>
        </ul>
      </div>
      <div
        class="socials"
      >
        <ul>
          <li>
            <i
              class="fa-brands fa-twitter"
            />
          </li>
          <li>
            <i
              class="fa-brands fa-facebook-f"
            />
          </li>
          <li>
            <i
              class="fa-brands fa-pinterest-p"
            />
          </li>
          <li>
            <i
              class="fa-brands fa-google-plus-g"
            />
          </li>
        </ul>
      </div>
      <div
        class="mobile cancel"
        role="button"
        tabindex="0"
      >
        <i
          class="fa-solid fa-xmark"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
